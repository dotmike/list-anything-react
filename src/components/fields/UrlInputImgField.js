import React from "react";
import PropTypes from "prop-types";
import { Form, Image } from "semantic-ui-react";

import InlineError from "../messages/InlineError";

const UrlInputImgField = ({ editable, error, value, field, onChange }) => {
  if (editable)
    return (
      <Form.Field error={!!error}>
        <Form.Input
          fluid
          type="url"
          label={field.description}
          value={value}
          onChange={onChange}
          name={field._id}
          placeholder="http://site.com/image.png"
        />
        {error && <InlineError text={error} />}
      </Form.Field>
    );
  return <Image size="tiny" centered src={value} />;
};

UrlInputImgField.propTypes = {
  editable: PropTypes.bool.isRequired,
  error: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  field: PropTypes.shape({
    _id: PropTypes.string,
    description: PropTypes.string
  }),
  onChange: PropTypes.func
};

UrlInputImgField.defaultProps = {
  field: {},
  onChange: () => {}
};

export default UrlInputImgField;
