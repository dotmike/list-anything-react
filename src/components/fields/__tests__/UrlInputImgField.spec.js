import UrlInputImgField from "../UrlInputImgField";

const testData = {
  value: "http://www.example.com/image.png",
  field: { _id: "1", description: "Testando:" },
  newValue: "http://google.com/image.png"
};

const setup = (propOverrides = {}) => {
  const props = {
    value: testData.value,
    error: "",
    editable: true,
    showFieldDescription: true,
    field: testData.field,
    onChange: jest.fn(),
    ...propOverrides
  };

  return {
    props,
    wrapperShallow: wrapperShallow(UrlInputImgField, props)
  };
};

describe("UrlInputImgField", () => {
  describe("when `editable` is true", () => {
    it("renders correctly", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: true
      });
      expect(wrapper()).toMatchSnapshot();
    });

    it("renders correctly when `error` is passed", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: true,
        error: "Something went wrong..."
      });
      expect(wrapper()).toMatchSnapshot();
    });

    it("calls `onChange` when changing the input value", () => {
      const { wrapperShallow: wrapper, props } = setup({
        editable: true
      });

      wrapper()
        .find("FormInput")
        .simulate("change", { target: { value: testData.newValue } });
      expect(props.onChange).toHaveBeenCalled();
    });
  });

  describe("when `editable` is false", () => {
    it("renders correctly when `showFieldDescription` is true", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: false,
        showFieldDescription: true
      });
      expect(wrapper()).toMatchSnapshot();
    });

    it("renders correctly when `showFieldDescription` is false", () => {
      const { wrapperShallow: wrapper } = setup({
        editable: false,
        showFieldDescription: false
      });
      expect(wrapper()).toMatchSnapshot();
    });
  });
});
