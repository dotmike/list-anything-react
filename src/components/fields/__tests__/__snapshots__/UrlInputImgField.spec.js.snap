// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlInputImgField when \`editable\` is false renders correctly when \`showFieldDescription\` is false 1`] = `
<Image
  as="img"
  centered={true}
  size="tiny"
  src="http://www.example.com/image.png"
  ui={true}
/>
`;

exports[`UrlInputImgField when \`editable\` is false renders correctly when \`showFieldDescription\` is true 1`] = `
<FormField>
  <b>
    Testando:
  </b>
  : 
  <Image
    as="img"
    size="tiny"
    src="http://www.example.com/image.png"
    ui={true}
  />
</FormField>
`;

exports[`UrlInputImgField when \`editable\` is true renders correctly 1`] = `
<FormField
  error={false}
>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    label="Testando:"
    name="1"
    onChange={[Function]}
    placeholder="http://site.com/image.png"
    type="url"
    value="http://www.example.com/image.png"
  />
</FormField>
`;

exports[`UrlInputImgField when \`editable\` is true renders correctly when \`error\` is passed 1`] = `
<FormField
  error={true}
>
  <FormInput
    as={[Function]}
    control={[Function]}
    fluid={true}
    label="Testando:"
    name="1"
    onChange={[Function]}
    placeholder="http://site.com/image.png"
    type="url"
    value="http://www.example.com/image.png"
  />
  <InlineError
    text="Something went wrong..."
  />
</FormField>
`;
